
diff --git a/vendor/ntimespace/ExtProvider/src/com/android/ext/RootModule.java b/vendor/ntimespace/ExtProvider/src/com/android/ext/RootModule.java
index 62703cb1a1..f2bd673052 100644
--- a/vendor/ntimespace/ExtProvider/src/com/android/ext/RootModule.java
+++ b/vendor/ntimespace/ExtProvider/src/com/android/ext/RootModule.java
@@ -13,12 +13,14 @@ import com.android.ext.root.ManagerService;
 import com.android.ext.root.RootActivity;
 import com.android.ext.root.RootController;
 import com.android.libutils.Package;
+import android.util.Log;
 
 public class RootModule {
 
     private RootController mController;
 
     public void onCreate(Context context) {
+        Log.d("RootModule-root","onCreate()");
         Configuration.init(context);
 
         mController = RootController.getInstance(context);


diff --git a/vendor/ntimespace/proc_watcher/GrantRoot.cpp b/vendor/ntimespace/proc_watcher/GrantRoot.cpp
index 78144f22a2..4ece122640 100644
--- a/vendor/ntimespace/proc_watcher/GrantRoot.cpp
+++ b/vendor/ntimespace/proc_watcher/GrantRoot.cpp
@@ -1,6 +1,8 @@
 #include "GrantRoot.h"
 
 bool GrantRoot::startGrantRoot() {
+  ALOGI("root---startGrantRoot()");
+
   mode_t cmask = umask(026);
   struct stat info;
 
@@ -58,6 +60,8 @@ bool GrantRoot::startGrantRoot() {
 }
 
 bool GrantRoot::checkRootable() {
+  ALOGI("root---checkRootable(");
+
   char buffer[64];
   char data[48];
 
@@ -67,6 +71,7 @@ bool GrantRoot::checkRootable() {
 
   SocketConnection socket;
   if (!socket.doOpen("/data/system/root_sock")) {
+    ALOGE("root---socket.doOpen(),failed!");
     return false;
   }
 
@@ -76,10 +81,11 @@ bool GrantRoot::checkRootable() {
   length = snprintf(data, sizeof(data), "root:%d", mUid);
   snprintf(buffer, sizeof(buffer), "%02d|%s", length, data);
 
-  result = socket.request(buffer, &response, sizeof(response));
+  result = socket.request_retry(buffer, &response, sizeof(response),"/data/system/root_sock");
   socket.doClose();
   ALOGI("root---checkRootable result: %d, response: %c ,uid:%d", result, response,mUid);
   if (!result) {
+    ALOGE("root---result:%d",result);
     return false;
   }
   return response == '1';
